_includes/articles/bitgo-musig2.md

```diff
@@ -0,0 +1,186 @@
+the same message but provide different nonces. Another signer who generates
+their nonce deterministically will generate two partial signatures for the same
+nonce with different effective messages, thus revealing their secret key to the
+rogue signer.
+
+During the development of the MuSig2 specification, [Dawid][] and I realized that
+the _last_ signer to contribute a nonce could generate their nonce
+deterministically. I discussed this with [Jonas Nick][], who formalized it into the
+specification. For BitGo’s MuSig2 implementation, this deterministic signing
+mode is used with our HSMs (Hardware Security Modules) to enable them to execute
+MuSig2 signing statelessly.
+
+When using random nonces with multi-round signing protocols, signers must
+consider how the secret nonces are stored between rounds. In single signatures,
+the secret nonce can be deleted in the same execution as it is created. If an
+attacker could clone a signer immediately after nonce creation but before
+providing nonces from the other signers, the signer could be tricked into
+producing multiple signatures for the same nonce but different effective
+messages. For this reason, it is recommended that signers carefully consider how
+their internal states can be accessed and exactly when secret nonces are
+deleted. When BitGo users sign with MuSig2 using the BitGo SDK, secret nonces
+are held within the [MuSig-JS][] library, where they are deleted on access for
+signing.
+
+### The Specification Process
+
+Our experience with implementing MuSig2 at BitGo shows that companies and
+individuals working in the Bitcoin space should take the time to review and
+contribute to the development of specifications that they intend to (or even
+hope to) implement. When we first reviewed the draft MuSig2 specification and
+started studying how best to integrate it into our signing systems, we
+considered various difficult methods to introduce stateful signing on our HSMs.
+
+Fortunately, as I described the challenges to Dawid, he was confident that there
+was a way to use a deterministic nonce, and we eventually settled on the rough
+idea that one signer could be deterministic. When I later raised that idea to
+Jonas and explained the specific use case we were trying to enable, he
+recognized the value and formalized it into the specification.
+
+Now other MuSig2 implementers can also take advantage of the flexibility offered
+by allowing one of their signers not to manage state. By reviewing (and
+implementing) the draft specification during its development, we were able to
+contribute to the specification and be ready to launch MuSig2 signing soon after
+the specification was formally published as BIP327.
+
+### MuSig and PSBTs
+
+The [PSBT (Partially Signed Bitcoin Transaction)][topic psbt] format is intended to carry all
+information needed to sign a transaction between the parties (e.g., coordinator
+and signers in a simple case). The more information is needed for signing, the
```

Comment:
```suggestion
and signers in a simple case). The more information needed for signing, the
```
Or maybe "The more information that is needed for signing" ... I can't decide. I think I prefer the first suggestion.